lib/AST/Stmt.cpp

```diff
@@ -36,7 +36,7 @@ using namespace swift;
 // Only allow allocation of Stmts using the allocator in ASTContext.
 void *Stmt::operator new(size_t Bytes, ASTContext &C,
                          unsigned Alignment) {
-  return C.Allocate(Bytes, Alignment);
+  return C.Allocate(Bytes, Alignment, AllocationArena::Declarations);
```

Comment:
Are you sure we want to allocate statements in the arena too?  Seems like it’s going to fragment the arena.